src/zoid/buttons/util.js

```diff
@@ -410,3 +413,29 @@ export const getModal: (
       });
   }
 });
+
+export const openPopupAtURI: (
+  uri: string,
+  merchantRequestedPopupsDisabled: ?boolean
+) => Object = (uri, merchantRequestedPopupsDisabled) => {
+  try {
+    if (userAgentSupportsPopups() && !merchantRequestedPopupsDisabled) {
+      return assertSameDomain(
+        popup(uri, {
+          width: DEFAULT_POPUP_SIZE.WIDTH,
+          height: DEFAULT_POPUP_SIZE.HEIGHT,
+        })
+      );
+    } else {
+      return window.open(uri, "_blank");
+    }
+  } catch (err) {
+    getLogger()
```

Comment:
I would suggest adding a fallback logic here to fallback to modal experience if popup fails.

Reply:
Thanks for the suggestion @ravishekhar. Sounds like we will be doing some dev work to make a purchase protection modal to address this issue that you and @wsbrunson called out.